File: Factorial-Python/factorial.py
def factorial(n):
    if n == 0:
        return 1
    else:
        return n * factorial(n-1)

def main():
	a = 50
	fact_a = factorial(a)
	print('factorial of {} is {}'.format(a, fact_a))

if __name__ == '__main__':
	main()

File: Factorial-Java/Factorial.java
import java.util.Scanner;

public class Factorial {

    private final static Scanner SC = new Scanner(System.in);

    /**
     * Gets the specified n's factorial.
     * @param n
     */
    private static int factorial(int n) {
        if (n == 0 || n == 1) {
            return n;
        }
        return n * factorial(n - 1);
    }

    public static void main(String[] args) {
        int n;
        
        System.out.print("Getting n's factorial. Enter n: ");
        n = SC.nextInt();
        
        System.out.println(String.format("%d's factorial is: %d", n, factorial(n)));
    }

}
